app/controllers/identity.js

```diff
@@ -14,16 +14,49 @@ export default class IdentityController extends Controller {
   @tracked isCopyClicked = false;
   @tracked isVerifyClicked = false;
   @tracked profileURL = this.model.profileURL || '';
+  @tracked saveDisabled = true;
+  @tracked generateChainCodeDisabled = this.model.profileURL === undefined;
+  @tracked checkboxDisabled =
+    this.generateChainCodeDisabled || this.model.chaincode === undefined;
+  @tracked linkDisabled = true;
 
   @action async handleRefresh() {
     window.location.reload();
   }
 
+  @action changeSaveDisabled() {
+    const isValidUrl = (urlString) => {
+      try {
+        return Boolean(new URL(urlString));
+      } catch (e) {
+        return false;
+      }
+    };
+    if (
+      this.profileURL === '' ||
+      this.profileURL === (this.model.profileURL || '') ||
+      !isValidUrl(this.profileURL)
+    ) {
+      this.saveDisabled = true;
+    } else {
+      this.saveDisabled = false;
+    }
+  }
+
+  @action changeLinkDisable() {
+    if (!this.isChecked) {
+      this.linkDisabled = false;
+    } else {
+      this.linkDisabled = true;
+    }
+  }
+
   @action handleCopy() {
     navigator.clipboard.writeText(this.Chaincode);
     this.isCopyClicked = true;
     if (this.isCopyClicked === true) {
       alert('Copied');
+      this.checkboxDisabled = false;
```

Comment:
Acc. to requirements link checkbox should be enabled when -  the user has a profile URL and chain-code in their user object. That means both conditions should be met. Are both conditions getting checked here? let me know if my understanding is wrong about the requirements

Reply:
those checks are getting done first in the above tracked variables but once the user click on generate chaincode and click on copy I want to show him the checkbox to ask him for a confirmation that he has added a chaincode and a profileURL